WIP - Improvements toward Addressing #173 (#240)
openjs-foundation/standards: MEMBER_EXPECTATIONS.md

# Working Group Member Expectations

> All participants in OpenJS Foundation projects and Working Groups must follow the [Code of Conduct](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/CODE_OF_CONDUCT.md). Standards Working Group Members are expected to participate in discussions related to OpenJS Foundation Projects, open source software, and the standards and specifications that bear on them. They are also expected educate and mentor newcomers to the field of standardization, lending support and advice to projects pursuing specification development. Finally, they are expected to help export OpenJS Foundation values to the standards communities they work with in order to make standards bodies nicer places to do work.  

Reviewer: ```suggestion
All participants in OpenJS Foundation projects and Working Groups must follow the [Code of Conduct](https://github.com/openjs-foundation/cross-project-council/blob/HEAD/CODE_OF_CONDUCT.md). Standards Working Group Members are expected to participate in discussions related to OpenJS Foundation Projects, open source software, and the standards and specifications that bear on them. They are also expected to educate and mentor newcomers to the field of standardization, lending support and advice to projects pursuing specification development. Finally, they are expected to help export OpenJS Foundation values to the standards communities they work with in order to make standards bodies nicer places to do work.  
```

There are further expectations for [OpenJS Foundation Representatives](https://github.com/openjs-foundation/standards/blob/main/MEMBER_REPRESENTATION.md) to standards and liaison organizations (hereafter called Representatives).

> ## Expectations for Representatives

Reviewer 2: Could we consider starting off by saying that consensus of the Standards WG allows a Representative to represent the position of OpenJSF as a whole on a particular topic? And that conversely, unless Rep has said consensus, you're not in a capacity to represent OpenJSF's position on that topic?

Reviewer 3: tl;dr: this would effectively prevent participation

This would be a very high bar to meet. Organization representatives are practically _always_ representing their organization's positions on topics.  For example, where participants in W3C working groups or delegates in Ecma committees represent an organization, those are the terms of their participation.  Strictly speaking, they do not participate in a personal capacity at all, so, at least in matters pertaining to normative topics, they are unable to represent a personal position, and as such are exclusively representing the position of their organization.

The OpenJS standards group does not have time to get consensus on all topics that may come up at standards bodies, and would very often find it difficult to even get a quorum of members that could speak to a given proposal.  We also typically only know topics one or two weeks in advance, if that.

Reviewer 3: having further read some of your comments further below, it sounds like you may be referring to things at, for example, the AC (W3C) or GA (Ecma) level rather than at the WG or TC level.  I think this section on member representation of the OpenJS standards group refers to participation in WGs/TCs.

in any case, it's good to be on the same page one way or the other

It is understood that Representatives may form opinions based on the needs of their employers, personal projects, or other experiences. While representing the Foundation, however, Representatives should participate based on what is in the best interest of the Foundation and its projects. If there is a conflict (e.g. between the Foundation, its projects, and/or personal or company positions), the Representative should be explicit about which perspective they are speaking from given the issue at hand. 

Representatives must also conduct themselves in a professional and respectful manner. General guidelines include:

> * Avoid invoking multiple "roles" when discussing challenging topics. This helps avoid [appeals to authority](https://en.wikipedia.org/wiki/Argument_from_authority) or the allusion that one's opinion is fully shared by multiple groups. 

Reviewer 3: This addresses the importance of avoiding "pulling rank" and misrepresenting others' positions.  You're right that there's nothing wrong with "documented consensus that you can point to" and this can help clarify things and/or prevent bikeshedding.

* Commit to ongoing development and governance best practices. Contribute good ideas, successful policies, better workflows, and other lessons learned back to the Working Group.
* Follow the contribution guidelines of the organization or project to which you are contributing.

### Building Consensus
Representatives may belong to one or more projects in the Foundation, and as such should feel free to discuss topics with each relevant group. In the case of conflicting opinions between Foundation projects, the Representative may raise an issue to the Standards Working Group for support. 

Representatives or Working Group Members may also call for an ad-hoc meeting to discuss controversial topics, or topics wherein a general consensus view has not yet emerged. Representatives are strongly encouraged to schedule ad-hoc meetings to refine thinking and inform stakeholders prior to plenary or liaison meetings where they will be discussed. Representatives should focus on developing and sharing the consenus of the Foundation and the projects impacted by the topic at hand. 

### Representatives and Decision-making

> Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should disclose any vote to the Standards Working Group, and provide a recommendation. If the Representative has an objection or plans to cast a dissenting vote, they should express it as early as possible to ensure there is ample time to discuss and reach consensus on the intended approach.

Reviewer 4: ```suggestion
Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should disclose any vote to the Standards Working Group, and provide a recommendation. If the Representative has an objection or plans to cast a dissenting vote while representing the Foundation, they should express it as early as possible to ensure there is ample time to discuss and reach consensus on the intended approach.
```

> Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should disclose any vote to the Standards Working Group, and provide a recommendation. If the Representative has an objection or plans to cast a dissenting vote, they should express it as early as possible to ensure there is ample time to discuss and reach consensus on the intended approach.

Reviewer 2: That a good step forward. I'd be a little more prescriptive:

```suggestion
Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should update the Standards Working Group of such opportunities that are relevant to the interests of OpenJSF or its projects well ahead of time so that the group is able to each consensus on the intended approach. The Representative must represent the group's decision and abstain in case the group hasn't been able to reach consensus.
```

I feel that this covers dissent (as the WG won't reach consensus in case the Rep disagrees with the majority).

> Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should disclose any vote to the Standards Working Group, and provide a recommendation. If the Representative has an objection or plans to cast a dissenting vote, they should express it as early as possible to ensure there is ample time to discuss and reach consensus on the intended approach.

Reviewer 4: I'm a bit confused.

While wearing the Foundation hat, we've already discussed and decided on how objections work - brought to Standards beforehand, and if it can't be for any reason, best judgement can be used and may need defending at the next Standards meeting.

While not representing the Foundation, an individual is under no obligation to interact with the Standards group about objections, even if they're a member of the Standards group.

> Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should disclose any vote to the Standards Working Group, and provide a recommendation. If the Representative has an objection or plans to cast a dissenting vote, they should express it as early as possible to ensure there is ample time to discuss and reach consensus on the intended approach.

Reviewer 2: I _think_ I understand the source of confusion.

Following some of the discussion and proposed changes in https://github.com/openjs-foundation/standards/pull/239, I was using the term _Representative_ in a much narrower way, i.e. to specifically mean when a person is representing OpenJSF (e.g. "@jorydotcom is OpenJSF's W3C _Representative_") but not when a person is operating in a working group in more of an _Invited Expert_ capacity.

Would it make sense to distinguish these roles more precisely? I think @rginn used the term _subject matter expert_ (SME) in a related issue which is more common than _Invited Expert_. Would that work for folks?

> Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should disclose any vote to the Standards Working Group, and provide a recommendation. If the Representative has an objection or plans to cast a dissenting vote, they should express it as early as possible to ensure there is ample time to discuss and reach consensus on the intended approach.

Reviewer 4: A person might be operating as a delegate of their employer while still being associated with OpenJS, and as such they don't need to answer to the Standards WG, so it's got nothing to do with IE/SME status imo.

> Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should disclose any vote to the Standards Working Group, and provide a recommendation. If the Representative has an objection or plans to cast a dissenting vote, they should express it as early as possible to ensure there is ample time to discuss and reach consensus on the intended approach.

Reviewer 2: > While wearing the Foundation hat, we've already discussed and decided on how objections work - brought to Standards beforehand, and if it can't be for any reason, best judgement can be used and may need defending at the next Standards meeting.

Sorry, I didn't see this documented anywhere. To be clear, I much prefer this as a solution. I think trusting that people will do the right thing and be proactive about building consensus if the topic feels more contentious is the right thing to do. However, I don't think it is acceptable for a _Representative_ of the OpenJSF in the narrow sense (i.e. not an _SME_), to vote against the consensus of the WG (and if the WG can't get to consensus, the Rep should abstain).

> While not representing the Foundation, an individual is under no obligation to interact with the Standards group about objections, even if they're a member of the Standards group.

100%. We should also make very clear that the persons is acting in an SME capacity in that case and not representing OpenJSF.

> Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should disclose any vote to the Standards Working Group, and provide a recommendation. If the Representative has an objection or plans to cast a dissenting vote, they should express it as early as possible to ensure there is ample time to discuss and reach consensus on the intended approach.

Reviewer 4: See #84 for previous discussion.

I completely agree with your two roles, and I think that's basically what #84 came to decide :-) we can bikeshed the terms later.

> Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should disclose any vote to the Standards Working Group, and provide a recommendation. If the Representative has an objection or plans to cast a dissenting vote, they should express it as early as possible to ensure there is ample time to discuss and reach consensus on the intended approach.

Reviewer 2: Awesome. I wasn't aware of that thread. Thanks for sharing. I don't really care what the terms are either. It would just be good if we could avoid them creating more confusion by overlapping with common terms already used by SDOs (e.g. delegate which means completely different things for Ecma and OSI).

> Some of OpenJS Foundation's memberships and liaison relationships permit Representatives to participate in their formal decision-making processes (e.g. voting). Representatives should disclose any vote to the Standards Working Group, and provide a recommendation. If the Representative has an objection or plans to cast a dissenting vote, they should express it as early as possible to ensure there is ample time to discuss and reach consensus on the intended approach.

Reviewer 3: it does seem like we need a mapping table 😄

* Provide liaison update reports to the Standards Working Group following liaison meetings. In the case that there are multiple Representatives participating in a meeting, only one needs to provide an update.
* Help develop and circulate positions on key issues from among relevant OpenJS Foundation projects and stakeholders. Where differences of opinion exist among Foundation participants, Representatives should work to establish a consensus opinion, and/or represent the opposing viewpoints as clearly as possible.

### Becoming a Representative
Community members who would like to officially represent the Foundation or their Project to an external organization should file an issue with the following information:
* The OpenJS Foundation's current relationship to this standards organization
* The scope of the work they plan to participate in (specific technical committees, proposals, interest groups, etc.)
* The relevance and potential impact of the work to OpenJS Foundation Projects
* Any participation costs (e.g. membership dues) needed

> Note that new liaisons or memberships must be discussed with OpenJS Foundation Executive Director Robin Ginn and must be approved by the Board of Directors.

Author: Update this: this group is empowered to decide whether to participate; ED will be required if a signature on legal agreement; Board approval may be required if $$ is required for the membership.
